github/pull_request.rb

```diff
@@ -179,9 +179,17 @@ def add_status(state, message, options = {})
     @client.create_status(repo.full_name, options[:sha], state, context: 'prprocessor', description: message, target_url: options[:url])
   end
 
-  def set_directory_labels(mapping)
+  def get_labels(filename, mapping)
+    mapping.select { |k, v| filename == k || filename.start_with?((k+"/").squeeze) }.values
```

Comment:
I didn't know squeeze but isn't this dangerous? Looks like it squeezes all duplicate characters but in a directory name you can have them for valid reasons (`lookups` would become `lokups`).

Reply:
My bad. I thought `squeeze` removes characters from the end of the string.
What I wanted to do is to remove extra `/` in the key. For example, if someone adds "app/views/" to the map then it probably won't add "app/views/" label because the file "app/views/models/index.html" doesn't start with "app/views//".

Maybe using `gsub` can be replaced here?